File: programmers/Lv_2/KimHyunSeo/탑.cpp
#include <string>
#include <vector>
#include <iostream>
using namespace std;
// 6 3 5 7 4
vector<int> solution(vector<int> heights)
{
	vector<int> answer = { 0 };
	int index = 0;
	int height = heights[0];
	for (int i = 1; i < heights.size(); i++)
	{
		if (heights[i - 1] > heights[i])
		{
			index = i;
			height = heights[i - 1];
		}
		if (height < heights[i]) {
			index = 0;
		}
		answer.push_back(index);
	}
	return answer;
}

int main()
{
	vector<int> answers = { 6,9,5,7,4 };
	solution(answers);
}

File: programmers/Lv_2/HwangJunghoon/탑.py
def solution(heights):
    #ex) heights=[1,5,3,6,7,6,5]
    answer = [0]*len(heights)
    # len(heights)==7, answer=[0,0,0,0,0,0,0]
    for i in range(len(heights)-1, 0, -1):
        for j in range(i-1, -1, -1):
            if heights[i]<heights[j]:
                answer[i]=j+1
                """
                [0,0,0,0,0,0,6]
                [0,0,0,0,0,5,6]
                ------------------(no addition : 2 times)
                [0,0,2,0,0,5,6]
                ------------------(no addition : 2 times)
                final : [0,0,2,0,0,5,6]
                """
                break
                # break 안하면 중복됨
    return answer